crates/sui-core/src/consensus_adapter.rs

```diff
@@ -367,23 +375,28 @@ impl ConsensusAdapter {
             "Submitting {:?} recovered pending consensus transactions to Narwhal",
             recovered.len()
         );
-        for transaction in recovered {
-            self.submit_unchecked(transaction, epoch_store);
-        }
+        self.submit_unchecked(recovered, epoch_store);
     }
 
     fn await_submit_delay(
         &self,
         committee: &Committee,
-        transaction: &ConsensusTransaction,
+        transactions: &Vec<ConsensusTransaction>,
```

Comment:
as I mentioned above, this can be `&[ConsensusTransaction]` so that either a vector or a temporary slice can be passed.